access/resource_permissions.go

```diff
@@ -353,7 +363,87 @@ func ResourcePermissions() *schema.Resource {
 		return nil
 	}
 	return &schema.Resource{
-		Schema:      s,
+		Schema: s,
+		CustomizeDiff: customdiff.Sequence(
+			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+				// Plan time validation for object permission levels
+				objectPermissionLevelMappingList := []struct {
+					objectKey               string
+					allowedPermissionLevels []string
+				}{
+					{
```

Comment:
Please incorporate with generic permissions mapping:

https://github.com/databrickslabs/terraform-provider-databricks/blob/993bc17c22cd32569fb3f65d071956c9722386a7/access/resource_permissions.go#L230

And perhaps fetch permission levels from API, so that provider won't have to be changed once new permissions are added on the platform

Reply:
Good idea, forgot that we had those on the API. Will do the improvement.

Reply:
However, one question just saw that the [API to get cluster permissions](https://docs.databricks.com/dev-tools/api/latest/permissions.html#operation/get-clusters-permission-levels) for instance requires a `cluster_id`. I'm thinking if this won't be a problem, since that the CustomizeDiff runs during the plan if I understood correctly there might be the case that the cluster_id is still unresolved and we won't be able to call the API. What are your thoughts here?

Reply:
@psg2  you're right. let's then keep them hardcoded. but in `permissionsResourceIDFields`

Reply:
Done.